src/main/java/com/mraof/minestuck/client/sounds/PlayerMusicClientHandler.java

```diff
@@ -0,0 +1,61 @@
+package com.mraof.minestuck.client.sounds;
+
+import com.mraof.minestuck.block.EnumCassetteType;
+import com.mraof.minestuck.network.MusicPlayerPacket;
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.resources.sounds.EntityBoundSoundInstance;
+import net.minecraft.client.sounds.SoundManager;
+import net.minecraft.sounds.SoundSource;
+import net.minecraft.world.entity.Entity;
+import net.minecraft.world.entity.player.Player;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Keep track of every player entities listening to the music.
+ * Create a sound instance on every client matching the music currently playing.
+ *
+ * @see MusicPlayerPacket
+ */
+
+public class PlayerMusicClientHandler
+{
+	private static final Map<Integer, EntityBoundSoundInstance> entitiesMap = new HashMap<>();
+	
+	public static void handlePacket(int entityID, EnumCassetteType cassetteType, float volume, float pitch)
+	{
+		checkEntitiesInMap();
+		
+		Entity entity = Minecraft.getInstance().level.getEntity(entityID);
+		if(entity instanceof Player)
+		{
+			SoundManager soundManager = Minecraft.getInstance().getSoundManager();
+			
+			if(entitiesMap.containsKey(entityID))
+			{
+				soundManager.stop(entitiesMap.remove(entityID));
```

Comment:
```suggestion
				//TODO: if the sound instance was started the same tick, it fails to stop properly
				soundManager.stop(entitiesMap.remove(entityID));
```
Leave a todo for problem 1 from a review comment on the item class.